app/components/files/file.tsx

```diff
@@ -66,74 +77,136 @@ const File = ({
         }
     }, [index]);
 
+    const handleOnOptionsPress = useCallback(() => {
+        if (onOptionsPress) {
+            onOptionsPress(index);
+        }
+    }, [index, onOptionsPress]);
+
+    const renderOptionsButton = useCallback(() => {
+        if (onOptionsPress) {
+            return (
+                <FileOptionsIcon
+                    onPress={handleOnOptionsPress}
+                />
+            );
+        }
+        return null;
+    }, [file, onOptionsPress, handleOnOptionsPress]);
+
+    const fileInfo = useMemo(() => {
+        return (
+            <FileInfo
+                file={file}
+                showDate={showDate}
+                channelName={channelName}
+                onPress={handlePreviewPress}
+                theme={theme}
+            />
+        );
+    }, [file, showDate, channelName, handlePreviewPress, theme]);
+
     const {styles, onGestureEvent, ref} = useGalleryItem(galleryIdentifier, index, handlePreviewPress);
 
-    if (isVideo(file) && publicLinkEnabled) {
+    const renderImageFileOverlay = useCallback(() => {
+        if (nonVisibleImagesCount) {
+            return (
+                <ImageFileOverlay
+                    theme={theme}
+                    value={nonVisibleImagesCount}
+                />
+            );
+        }
+        return null;
+    }, [theme, nonVisibleImagesCount]);
+
+    const imageFile = useMemo(() => {
```

Comment:
It doesn't seem needed to keep in memory the three different possible file representations. Maybe one single useMemo to hold them all?

That being said... probably useMemo is not needed to store a component itself, since React should have its own ways to recognize if it is the same component or not. Therefore... instead of useMemo, you can use something like:
```
let fileComponent;
if (isVideo(file) {
    component = ...
} else if (isDocument(file) {
    component = ...
} ...
```

Reply:
I cleaned this up quite a bit.  Any function that only returns a component is no longer memoized (left up to React).